### ISSUE TITLE
amdgpu disassembler ignores opsel on vop3 instructions

### ISSUE THREAD

**Issue by DadSchoorse** [llvm/llvm-project#54119]
```
Representation: Assembly (Final Assembly)

BB0:
    s_mov_b32 s0, s3                                            ; be800303
    s_movk_i32 s3, 0x8000                                       ; b0038000
    s_load_dwordx8 s[8:15], s[2:3], 0xe0                        ; f40c0201 fa0000e0
    v_lshl_add_u32 v2, s0, 3, v0                                ; d7460002 04010600
    v_lshl_add_u32 v3, s4, 3, v1                                ; d7460003 04050604
    v_add_nc_u16 v0, v2, 4                                      ; d7030000 00010902
    v_add_nc_u16 v0, v3, 4                                      ; d7034000 00010903
    v_pack_b32_f16 v1, v0, 0xfd66                               ; d7110001 0001ff00 0000fd66
    s_waitcnt lgkmcnt(0)                                        ; bf8cc07f
    image_store v[0:1], v[2:3], s[8:15] dmask:0xf dim:SQ_RSRC_IMG_2D unorm glc d16 ; f0203f08 80020002
    s_endpgm                                                    ; bf810000
```

This was created by radv+aco using llvm 13.0.1 as the disassembler, and I think the second v_add_nc_u16 is using opsel to write to the high half of v0, but that's not visible in the resulting assembly.

**Comment by dpreobra** [comment]
I must note that this is not a disassembler issue. op_sel is not supported yet for v_add_nc_u16 (GFX10) but it should.

**Comment by dpreobra** [comment]
See Issue https://github.com/llvm/llvm-project/issues/54793 for a full list of GFX10 opcodes which have missing op_sel support.

**Comment by dpreobra** [comment]
Fixed by commit ab18e1a533c3ca69597f9da319f943361b157d88
